### PR TITLE
imlib/apriltag: Disable unused TAG families by default.

### PR DESCRIPTION
**Pull request by kwagyeman**
Disable extra tag families that no one uses. This saves flash and makes the code run faster as you don't have to search through unused tag families. The IDE recommends TAG36H11 right now. The other families should generally be avoided anyway.

### CODE REVIEW

File: scripts/examples/06-April-Tags/find_apriltags_max_res.py [hunk 1]
```diff
@@ -31,47 +30,15 @@
...
     clock.tick()
     img = sensor.snapshot()
-    for tag in img.find_apriltags(
-        families=tag_families
-    ):  # defaults to TAG36H11 without "families".
+    for tag in img.find_apriltags():
         img.draw_rectangle(tag.rect(), color=127)
         img.draw_cross(tag.cx(), tag.cy(), color=127)
-        print_args = (family_name(tag), tag.id(), (180 * tag.rotation()) / math.pi)
+        print_args = ("TAG36H11", tag.id(), (180 * tag.rotation()) / math.pi)
```
Comment (iabdalkader) [inline scripts/examples/06-April-Tags/find_apriltags_max_res.py:42]:
```suggestion
        print_args = ("TAG36H11", tag.id(), (180 * tag.rotation()) / math.pi)
```
But some boards will have other families enabled, maybe return the name with or `tag.family` ? Also converting all of those to attributes will probably save lots of space.

File: src/omv/boards/ARDUINO_GIGA/imlib_config.h [hunk 2]
```diff
@@ -96,6 +96,12 @@
 
 // Enable find_apriltags() (64 KB)
 #define IMLIB_ENABLE_APRILTAGS
+//#define IMLIB_ENABLE_APRILTAGS_TAG16H5
+//#define IMLIB_ENABLE_APRILTAGS_TAG25H7
+//#define IMLIB_ENABLE_APRILTAGS_TAG25H9
+//#define IMLIB_ENABLE_APRILTAGS_TAG36H10
+#define IMLIB_ENABLE_APRILTAGS_TAG36H11
+//#define IMLIB_ENABLE_APRILTAGS_ARTOOLKIT
```
Comment (iabdalkader) [inline src/omv/boards/ARDUINO_GIGA/imlib_config.h:104]:
```suggestion
#define IMLIB_ENABLE_APRILTAGS_ARTOOLKIT
```
I'd leave this one too by default for all boards, as it's popular.

File: src/omv/boards/ARDUINO_GIGA/imlib_config.h [hunk 3]
```diff
@@ -96,6 +96,12 @@
 
 // Enable find_apriltags() (64 KB)
 #define IMLIB_ENABLE_APRILTAGS
+//#define IMLIB_ENABLE_APRILTAGS_TAG16H5
+//#define IMLIB_ENABLE_APRILTAGS_TAG25H7
+//#define IMLIB_ENABLE_APRILTAGS_TAG25H9
+//#define IMLIB_ENABLE_APRILTAGS_TAG36H10
```
Comment (iabdalkader) [inline src/omv/boards/ARDUINO_GIGA/imlib_config.h:102]:
```suggestion
```
Let's just define the enabled ones, until we have a common `imlib_config.h`

**Review by iabdalkader — CHANGES_REQUESTED** [review]

**Comment by iabdalkader** [comment]
@kwagyeman Let's get this change merged and then do a release.

File: src/omv/boards/ARDUINO_GIGA/imlib_config.h (hunk 2 above)
Comment (kwagyeman) [reply src/omv/boards/ARDUINO_GIGA/imlib_config.h:104]: I've never seen people to use it. Sure? This isn't the same as the ArUco codes that OpenCV works with. I think it's just a old tag formt that AprilTags was using. It's #2 of this image: 
image

File: scripts/examples/06-April-Tags/find_apriltags_max_res.py (hunk 1 above)
Comment (kwagyeman) [reply scripts/examples/06-April-Tags/find_apriltags_max_res.py:42]: Should this be in another PR? Providing the family name as a function is a missing feature. I could make it as an attribute, but, converting all those over should be done at the same time in a release with other breaking changes I think.

File: src/omv/boards/ARDUINO_GIGA/imlib_config.h (hunk 3 above)
Comment (kwagyeman) [reply src/omv/boards/ARDUINO_GIGA/imlib_config.h:102]: Okay

File: scripts/examples/06-April-Tags/find_apriltags_max_res.py (hunk 1 above)
Comment (iabdalkader) [reply scripts/examples/06-April-Tags/find_apriltags_max_res.py:42]: Okay but the script can use that attribute now.

Comment (kwagyeman) [reply scripts/examples/06-April-Tags/find_apriltags_max_res.py:42]: Okay, will make an attribute with the family name string.

File: src/omv/boards/ARDUINO_GIGA/imlib_config.h (hunk 2 above)
Comment (iabdalkader) [reply src/omv/boards/ARDUINO_GIGA/imlib_config.h]: No I'm not sure, it just sounded familiar. Okay I guess the default one is fine.

File: scripts/examples/06-April-Tags/find_apriltags_max_res.py (hunk 1 above)
Comment (iabdalkader) [reply scripts/examples/06-April-Tags/find_apriltags_max_res.py:42]: Actually you should convert them now while we're at it, this way we can see the total size savings ?

Comment (kwagyeman) [reply scripts/examples/06-April-Tags/find_apriltags_max_res.py]: Done, added changed in another commit for this. But, I cannot test this while on the road.

Comment (kwagyeman) [reply scripts/examples/06-April-Tags/find_apriltags_max_res.py]: Looks like ~224 bytes or so. I deleted the subscr function though which probably accounts for those savings. Moving to attribute seems to be a minor savings.

File: src/omv/modules/py_image.c [hunk 4]
```diff
@@ -5096,190 +5096,123 @@ static void py_apriltag_print(const mp_print_t *print, mp_obj_t self_in, mp_prin
...
+                dest[0] = self->h;
+                break;
+            case MP_QSTR_id:
+                dest[0] = self->id;
+                break;
+            case MP_QSTR_family:
+                dest[0] = self->family;
+                break;
+            case MP_QSTR_family_str:
+                #ifdef IMLIB_ENABLE_APRILTAGS_TAG16H5
```
Comment (iabdalkader) [inline src/omv/modules/py_image.c:5129]: There's no need for this `apriltag_families` are always defined, a switch statement here is fine even if not all of them are enabled.

File: src/omv/modules/py_image.c [hunk 5]
```diff
@@ -5096,190 +5096,123 @@ static void py_apriltag_print(const mp_print_t *print, mp_obj_t self_in, mp_prin
...
+                if (mp_obj_get_int(self->family) == TAG16H5) {
+                    dest[0] = "TAG16H5";
```
Comment (iabdalkader) [inline src/omv/modules/py_image.c:5131]: You can't return a const chat here of course, you need an mp_obj there's a str from cstr function somewhere.

File: scripts/examples/06-April-Tags/find_apriltags_w_lens_zoom.py [hunk 6]
```diff
@@ -32,8 +32,8 @@
     clock.tick()
     img = sensor.snapshot()
     for tag in img.find_apriltags():  # defaults to TAG36H11
-        img.draw_rectangle(tag.rect(), color=(255, 0, 0))
-        img.draw_cross(tag.cx(), tag.cy(), color=(0, 255, 0))
-        print_args = (tag.id(), (180 * tag.rotation()) / math.pi)
-        print("Tag Family TAG36H11, Tag ID %d, rotation %f (degrees)" % print_args)
+        img.draw_rectangle(tag.rect, color=(255, 0, 0))
+        img.draw_cross(tag.cx, tag.cy, color=(0, 255, 0))
+        print_args = (tag.family_str, tag.id, (180 * tag.rotation) / math.pi)
```
Comment (iabdalkader) [inline scripts/examples/06-April-Tags/find_apriltags_w_lens_zoom.py:37]: How about `tag.name` ?

**Review by iabdalkader — CHANGES_REQUESTED** [review]

File: scripts/examples/06-April-Tags/find_apriltags_max_res.py (hunk 1 above)
Comment (iabdalkader) [reply scripts/examples/06-April-Tags/find_apriltags_max_res.py]: Yeah I noticed that too was wondering why its needed. Less code is better anyway.

File: src/omv/modules/py_image.c (hunk 5 above)
Comment (kwagyeman) [reply src/omv/modules/py_image.c:5131]: Doh! Yes, of course.

Comment (kwagyeman) [reply src/omv/modules/py_image.c:5131]: Like

File: src/omv/modules/py_image.c (hunk 4 above)
Comment (kwagyeman) [reply src/omv/modules/py_image.c]: Done

**Comment by kwagyeman** [comment]
I went through and tested all the scripts using AprilTags.

**Review by iabdalkader — APPROVED** [review]

**Comment by iabdalkader** [comment]
In the future, changes like this should have the breaking label. I'll then collect and add them the stable release log.

**Pull request merged**